Tests/IntegrationTests/Infrastructure/PluginInspector.cs

```diff
@@ -0,0 +1,215 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="PluginInspector.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarQube.Common;
+using SonarQube.Plugins.Common;
+using SonarQube.Plugins.Maven;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+
+namespace SonarQube.Plugins.IntegrationTests
+{
+    /// <summary>
+    /// Wrapper around a Java class that loads a plugin from a jar and extracts information
+    /// from it. The plugin information is returned as an XML file
+    /// </summary>
+    internal class PluginInspector
+    {
+        private const string PluginInspectorFullClassName = "PluginInspector";
+
+        private readonly IJdkWrapper jdkWrapper;
+        private readonly ILogger logger;
+
+        private readonly string tempDir;
+        private string inspectorClassFilePath;
+        
+        public PluginInspector(ILogger logger, string tempDir)
+        {
+            if (logger == null)
+            {
+                throw new ArgumentNullException("logger");
+            }
+            if (string.IsNullOrWhiteSpace(tempDir))
+            {
+                throw new ArgumentNullException("tempDir");
+            }
+
+
+            this.logger = logger;
+            this.tempDir = tempDir;
+            this.jdkWrapper = new JdkWrapper();
+        }
+
+        public JarInfo GetPluginDescription(string jarFilePath)
+        {
+            Assert.IsTrue(File.Exists(jarFilePath), "Jar file does not exist");
+
+            this.Build(jarFilePath);
+
+            string reportFilePath = this.RunPluginInspector(jarFilePath, tempDir);
+
+            JarInfo jarInfo = null;
+            if (reportFilePath != null)
+            {
+                jarInfo = JarInfo.Load(reportFilePath);
+            }
+
+            return jarInfo;
+        }
+
+        private void CheckJdkIsInstalled()
+        {
+            if (!this.jdkWrapper.IsJdkInstalled())
+            {
+                Assert.Inconclusive("Test requires the JDK to be installed");
+            }
+        }
+
+        private void Build(string jarFilePath)
+        {
+            Assert.IsTrue(File.Exists(jarFilePath), "Jar file does not exist");
+
+            // Get the java source files
+            Directory.CreateDirectory(this.tempDir);
+            string srcDir = CreateSubDir(this.tempDir, "src");
+            string outDir = CreateSubDir(this.tempDir, "out");
+            string xxxDir = CreateSubDir(this.tempDir, "xxx");
```

Comment:
TODO: check whether the underlying compiler wrapper actually needs this directory: it seems to always be empty.